--- luabind/detail/call_function.hpp.old
+++ luabind/detail/call_function.hpp
@@ -323,7 +323,8 @@
 
 #endif // LUABIND_CALL_FUNCTION_HPP_INCLUDED
 
-#elif BOOST_PP_ITERATION_FLAGS() == 1
+#else
+    #if BOOST_PP_ITERATION_FLAGS() == 1
 
 #define LUABIND_TUPLE_PARAMS(z, n, data) const A##n *
 #define LUABIND_OPERATOR_PARAMS(z, n, data) const A##n & a##n
@@ -439,5 +440,6 @@
 #undef LUABIND_TUPLE_PARAMS
 
 
+    #endif
 #endif
 
--- luabind/detail/call_member.hpp.old
+++ luabind/detail/call_member.hpp
@@ -316,7 +316,8 @@
 
 #endif // LUABIND_CALL_MEMBER_HPP_INCLUDED
 
-#elif BOOST_PP_ITERATION_FLAGS() == 1
+#else
+    #if BOOST_PP_ITERATION_FLAGS() == 1
 
 #define LUABIND_TUPLE_PARAMS(z, n, data) const A##n *
 #define LUABIND_OPERATOR_PARAMS(z, n, data) const A##n & a##n
@@ -359,5 +360,6 @@
 #undef LUABIND_OPERATOR_PARAMS
 #undef LUABIND_TUPLE_PARAMS
 
+    #endif
 #endif
 
--- luabind/wrapper_base.hpp.old
+++ luabind/wrapper_base.hpp
@@ -89,7 +89,8 @@
 
 #endif // LUABIND_WRAPPER_BASE_HPP_INCLUDED
 
-#elif BOOST_PP_ITERATION_FLAGS() == 1
+#else
+    #if BOOST_PP_ITERATION_FLAGS() == 1
 
 #define LUABIND_TUPLE_PARAMS(z, n, data) const A##n *
 #define LUABIND_OPERATOR_PARAMS(z, n, data) const A##n & a##n
@@ -143,7 +144,8 @@
 #undef LUABIND_OPERATOR_PARAMS
 #undef LUABIND_TUPLE_PARAMS
 
-#else // free call_member forwardarding functions
+    #else
+//#else // free call_member forwardarding functions
 
 #define N BOOST_PP_ITERATION()
 
@@ -187,4 +189,5 @@
 
 #undef N
 
+    #endif
 #endif

